src/eko/io/types.py

```diff
@@ -0,0 +1,174 @@
+"""Common type definitions, only used for static analysis.
+
+Unfortunately, Python has still some discrepancies between runtime classes and
+type hints, so it is better not to mix dataclasses and generics.
+
+E.g. before it was implemented::
+
+    @dataclasses.dataclass
+    class RunningReference(DictLike, Generic[Quantity]):
+        value: Quantity
+        scale: float
+
+but in this way it is not possible to determine that ``RunningReference`` is
+subclassing ``DictLike``, indeed::
+
+    inspec.isclass(RunningReference)  # False
+    issubclass(RunningReference, DictLike)  # raise an error, since
+                                            # RunningReference is not a class
+
+Essentially classes can be used for type hints, but types are not all classes,
+especially when they involve generics.
+
+For this reason I prefer the less elegant dynamic generation, that seems to
+preserve type hints.
+
+"""
+import dataclasses
+import enum
+import typing
+from math import isnan
+from typing import Any, Dict, Optional
+
+from .dictlike import DictLike
+
+T = typing.TypeVar("T")
+
+
+def reference_running(quantity: typing.Type[typing.Union[int, float]]):
+    """Generate running quantities reference point classes.
+
+    The motivation for dynamic generation is provided in module docstring.
+
+    """
+
+    @dataclasses.dataclass
+    class ReferenceRunning(DictLike):
+        value: quantity
+        scale: float
+
+    return ReferenceRunning
+
+
+IntRef = reference_running(int)
+FloatRef = reference_running(float)
+
+Order = typing.Tuple[int, int]
+FlavorsNumber = int
+FlavorIndex = int
+IntrinsicFlavors = typing.List[FlavorIndex]
+
+
+@dataclasses.dataclass
+class CouplingsRef(DictLike):
+    """Reference values for coupling constants."""
+
+    alphas: FloatRef
+    alphaem: FloatRef
+    num_flavs_ref: Optional[FlavorsNumber] = None
```

Comment:
do we really want a default here? (and below?)

Reply:
Already declared, there is an open point in the OP. It was just an easy fix.